### PR TITLE
Add Splunk .NET docker image to the chart

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Add Splunk .NET docker image to the chart. Add to rotation for image updates.

### CODE REVIEW

File: helm-charts/splunk-otel-collector/values.yaml [hunk 1]
```diff
@@ -1179,6 +1179,9 @@ operator:
       nodejs:
         repository: ghcr.io/signalfx/splunk-otel-js/splunk-otel-js
         tag: v2.6.0
+      dotnet:
+        repository: ghcr.io/signalfx/splunk-otel-dotnet/splunk-otel-dotnet
+        tag: 1.2.0
```
Comment (pjanotti) [inline helm-charts/splunk-otel-collector/values.yaml:1184]: Depends on https://github.com/signalfx/splunk-otel-dotnet/pull/397 - the tag should be `1.2.1`

**Review by pjanotti — COMMENTED** [review]
LGTM, just need to wait for us to publish the Docker image.

**Review by Kielek — COMMENTED** [review]
I think that you are missing, one important thing.
Operator has to setup env. var.: `"OTEL_DOTNET_AUTO_PLUGINS": "Splunk.OpenTelemetry.AutoInstrumentation.Plugin, Splunk.OpenTelemetry.AutoInstrumentation"`

Otherwise it will be working us Vanilla/upstream distribution.

There will be no support for RUM, realms/splunk access key, AlwaysOn Profiling (in progress), come gdi-config requierments.

File: .chloggen/adddotnetimage.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+# The name of the component, or a single word describing the area of concern, (e.g. agent, clusterReceiver, gateway, operator, chart, other)
+component: operator
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Use the Splunk Otel .NET docker image by default when auto-instrumenting with the operator
```
Comment (Kielek) [inline .chloggen/adddotnetimage.yaml:6]:
```suggestion
note: Use the Splunk Distribution of OpenTelemetry .NET docker image by default when auto-instrumenting with the operator
```

**Review by Kielek — APPROVED** [review]
Changes looks good to me, but didn't have chance to test it E2E. I hope you have did it.

Two more thing
1. Should we update our procedure to include changes in this repository? https://github.com/signalfx/splunk-otel-dotnet/blob/608f45e960eb6f6af55c0554a21e40527f20faa8/RELEASING.md Or it willl be on your side? What is the typical scenario?
2. Minor comment related to the changelog entry.

**Comment by atoulme** [comment]
> Should we update our procedure to include changes in this repository? https://github.com/signalfx/splunk-otel-dotnet/blob/608f45e960eb6f6af55c0554a21e40527f20faa8/RELEASING.md Or it willl be on your side? What is the typical scenario?
We run a cron job to pick up new versions, and open a PR automatically. We do this for java and nodejs already. This PR adds the dotnet image to the rotation.

File: helm-charts/splunk-otel-collector/values.yaml [hunk 3]
```diff
@@ -1180,6 +1180,12 @@ operator:
       nodejs:
         repository: ghcr.io/signalfx/splunk-otel-js/splunk-otel-js
         tag: v2.6.1
+      dotnet:
+        repository: ghcr.io/signalfx/splunk-otel-dotnet/splunk-otel-dotnet
+        tag: 1.2.1
```
Comment (jvoravong) [inline helm-charts/splunk-otel-collector/values.yaml:1185]: https://github.com/signalfx/splunk-otel-dotnet/pkgs/container/splunk-otel-dotnet%2Fsplunk-otel-dotnet
```suggestion
        tag: v1.2.1
```

**Comment by atoulme** [comment]
Should we add a test application similar to node.js? @jvoravong wdyt?

**Review by pjanotti — APPROVED** [review]

**Comment by jvoravong** [comment]
Was able to verify the Splunk .NET docker image works e2e. There is an issue in .NET environments where the exporter endpoint isn't always respected properly which I will address in https://github.com/signalfx/splunk-otel-collector-chart/pull/1117.

Merging this PR now.

**Pull request merged**